src/commands/synthetics/api.ts

```diff
@@ -259,50 +266,69 @@ const createMobileVersion = (request: (args: AxiosRequestConfig) => AxiosPromise
   return resp.data
 }
 
-type RetryPolicy = (retries: number, error: AxiosError) => number | undefined
+const retryWithJitter = (delay: number = DELAY_BETWEEN_429_RETRIES) => delay + Math.floor(Math.random() * delay)
+
+export type RetryPolicy = {
+  retryOn404?: boolean | undefined
+  retryOn429?: boolean | undefined
+}
 
-const retryOn5xxErrors: RetryPolicy = (retries, error) => {
-  // Retry on Node.js errors for both retry policies.
+export const determineRetryDelay = (
+  retries: number,
+  error: Error,
+  retryPolicy: RetryPolicy = {retryOn404: false, retryOn429: false}
+) => {
+  // Always retry on Node.js errors
   if (retries < MAX_RETRIES && isNodeError(error)) {
     return LARGE_DELAY_BETWEEN_RETRIES
   }
 
+  // Always retry on 5xx
   if (retries < MAX_RETRIES && is5xxError(error)) {
     return DELAY_BETWEEN_RETRIES
   }
-}
 
-const retryOn5xxOr404Errors: RetryPolicy = (retries, error) => {
-  const retryOn5xxDelay = retryOn5xxErrors(retries, error)
-  if (retryOn5xxDelay) {
-    return retryOn5xxDelay
+  // Retry on 404
+  if (retryPolicy.retryOn404 && retries < MAX_RETRIES && isNotFoundError(error)) {
+    return DELAY_BETWEEN_RETRIES
   }
 
-  if (retries < MAX_RETRIES && isNotFoundError(error)) {
-    return DELAY_BETWEEN_RETRIES
+  // Retry on 429
+  if (retryPolicy.retryOn429 && retries < MAX_RETRIES && isTooManyRequestsError(error)) {
+    return retryWithJitter(DELAY_BETWEEN_429_RETRIES)
   }
 }
 
-const getErrorHttpStatus = (error: AxiosError | EndpointError) =>
-  'status' in error ? error.status : error.response?.status
+const getErrorHttpStatus = (error: Error): number | undefined =>
+  'status' in error
+    ? error.status
+    : 'response' in error && 'status' in (error.response as any)
+    ? (error.response as any)?.status
+    : undefined
```

Comment:
You don't need any `any`: TypeScript generates a "pseudo-type" for you 😁 
```suggestion
const getErrorHttpStatus = (error: Error): number | undefined =>
  'status' in error
    ? error.status
    : 'response' in error && 'status' in error.response
    ? error.response.status
    : undefined
```

Reply:
I was hoping so as well, but it really doesn't like it without it 😅

Reply:
Yeah nah it doesn't like it :D 
https://github.com/DataDog/datadog-ci/actions/runs/8357462396/job/22876652674?pr=1239

Reply:
Ha, yes it's because we are missing the type of `error.status`, `error.response`, and `error.response.status`.

Reply:
Then you can do something like that instead:

```ts
const isEndpointError = (error: Error): error is EndpointError => error instanceof EndpointError

const getErrorHttpStatus = (error: Error): number | undefined =>
  isAxiosError(error) ? error.response?.status : isEndpointError(error) ? error.status : undefined
```

which requires the following imports:
```ts
import type {AxiosError, AxiosPromise, AxiosRequestConfig} from 'axios'

import {isAxiosError} from 'axios'
```

Reply:
I like how clean it looks like this, but it doesn't work 😅 tests fail since it doesn't do the correct number of retries. Will try and see if i can massage it into working 👍